modules/ROOT/pages/mule-expression-language-tips.adoc

```diff
@@ -390,7 +390,19 @@ Produces a NullPointer exception
 +
 Note also that, if given the expression `#[flowVars.abc.toString()]` and the value of '`abc`' is `null`, Mule throws a `NullPointerException`.
 
+=== Evaluation Gotchas
+
+There is a known issue that causes some incorrect expressions to work during an evaluation without causing any compiler errors. The issue occurs when an unexpected character is affixed at the end of an escaped property in an expression that is part of an evaluation, for example:
+
+```
+<when expression="#[message.inboundProperties.'http.request.path'] == '/some_path']">
+```
+
+The previous expression is incorrect, because a `]` character is incorrectly affixed to the `'http.request.path'` property. However, Mule ignores this extra character and is able to compile the expression without errors.
+
+Keep in mind this parsing issue while troubleshooting incorrect expressions or if you experience unexpected behaviors as a result of evaluating an expression. This behavior is present in all Mule 3 versions, in any component that evaluates a MEL expression.
+
 == See Also
 
 * For the complete MEL reference, including lists of operators, imported Java classes, context objects, etc. see xref:mule-expression-language-reference.adoc[Mule Expression Language Reference].
-* Quick tips for MEL xref:mule-expression-language-tips.adoc[Mule Expression Language Tips]
\ No newline at end of file
+* Quick tips for MEL xref:mule-expression-language-tips.adoc[Mule Expression Language Tips]
```

Comment:
* For a quick reference to MEL, see xref:mule-expression-language-tips.adoc[Mule Expression Language Tips].